optaplanner-persistence/optaplanner-persistence-jackson/src/main/java/org/optaplanner/persistence/jackson/api/score/DeserializationUtils.java

```diff
@@ -0,0 +1,30 @@
+package org.optaplanner.persistence.jackson.api.score;
+
+import java.io.IOException;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+public class DeserializationUtils {
+
+    @SuppressWarnings("unchecked")
+    public static <T> T deserializePolymorphicProperty(DeserializationContext context, JsonNode tree, String property)
+            throws ClassNotFoundException, JsonParseException, JsonMappingException, IOException {
+        if (!tree.has(property) || !tree.has(property + "Class")) {
+            throw new IllegalStateException("Tree does not have specified property and it class infomation! Tree:\n"
+                    + tree.toString());
+        }
+        ObjectMapper mapper = new ObjectMapper();
```

Comment:
A Deserializer that creats an ObjectMapper?? That that doesn't make sense?

Reply:
Fixed; learned about parser.getCodec() which supplies the ObjectMapper the parser using.